src/backend/distributed/commands/dependencies.c

```diff
@@ -135,6 +135,45 @@ EnsureDependenciesExistOnAllNodes(const ObjectAddress *target)
 }
 
 
+/*
+ * GetDistributableDependenciesForObject finds all the dependencies that Citus
+ * can distribute and returns those dependencies regardless of their existency
+ * on nodes.
+ */
+List *
+GetDistributableDependenciesForObject(const ObjectAddress *target)
+{
+	/* local variables to work with dependencies */
+	List *distributableDependencies = NIL;
+	ListCell *dependencyCell = NULL;
+
+	/* collect all dependencies in creation order */
+	List *dependencies = GetDependenciesForObject(target);
+
+	/* filter the ones that can be distributed */
+	foreach(dependencyCell, dependencies)
+	{
+		ObjectAddress *dependency = (ObjectAddress *) lfirst(dependencyCell);
+
+		/*
+		 * TODO: maybe we can optimize the logic applied in below line. Actually we
+		 * do not need to create ddl commands as we are not ensuring their existence
+		 * in nodes, but we utilize logic it follows to choose the objects that could
+		 * be distributed
+		 */
+		List *dependencyCommands = GetDependencyCreateDDLCommands(dependency);
+
+		/* create a new list with dependencies that actually created commands */
```

Comment:
I think this should be moved one line below maybe as it doesnt explain this if check but what is done inside if check?

Also I would remove this comment as we already see that we are appending to the list clearly by the function name `lappend`.